ethcore/src/verification/queue/mod.rs

```diff
@@ -681,6 +681,12 @@ impl<K: Kind> Drop for VerificationQueue<K> {
 		*self.state.0.lock() = State::Exit;
 		self.state.1.notify_all();
 
+		// acquire this lock to force threads to reach the waiting point
+		// if they're in-between the exit check and the more_to_verify wait.
+		{
+			let _more = self.verification.more_to_verify.lock().unwrap();
+		}
+
 		// wake up all threads waiting for more work.
 		self.more_to_verify.notify_all();
```

Comment:
i think that would cause a deadlock in itself, since any notified thread needs to have the lock returned from the wait call.